packages/system/src/components/comparison-table/index.tsx

```diff
@@ -59,31 +79,103 @@ export function ComparisonTable({
 			})
 			setData(data)
 		}
-		fetchData()
-			.then(() => handleSort("name"))
-			.catch((error) => {
-				if (error.name === "AbortError") {
-					return
-				}
-			})
+
+		if (!mountedRef.current) {
+			mountedRef.current = true
+
+			fetchData()
+				.then(() => {
+					handleSort("name")
+				})
+				.catch((error) => {
+					if (error.name === "AbortError") {
+						return
+					}
+				})
+				.finally(() => {
+					setLoading(false)
+				})
+		}
+
 		return () => abortController.abort()
-	}, [libraries])
+	}, [handleSort, libraries])
 
 	const libraryStats = React.useMemo(
 		() => sortLibraries(data, sortConfig),
 		[data, sortConfig]
 	)
 
-	function handleSort(heading: typeof sortConfig.by) {
-		setSortConfig((prevSort) => {
-			if (prevSort.by === heading) {
-				return { by: heading, asc: !prevSort.asc }
-			}
-			if (heading === "name" || heading === "author") {
-				return { by: heading, asc: true }
-			}
-			return { by: heading, asc: false }
-		})
+	function handleTableRows() {
```

Comment:
Might be worth factoring out the table body and the skeleton into their own components and import them, using the logic in this component to decide which to render. Should shorten this file and make it a bit easier to read.